internal/olm/operator/registry/index_image.go

```diff
@@ -349,3 +714,67 @@ func (c IndexImageCatalogCreator) deleteRegistryPod(ctx context.Context, podName
 
 	return nil
 }
+
+// StringifyDecConfig writes the generated declarative config to a string.
+func StringifyDecConfig(declcfg *declarativeconfig.DeclarativeConfig) (string, error) {
+	var buf bytes.Buffer
+	err := declarativeconfig.WriteYAML(*declcfg, &buf)
+	if err != nil {
+		return "", fmt.Errorf("error writing generated declarative config to JSON encoder: %v", err)
+	}
+
+	return buf.String(), nil
+}
+
+// ValidateFBC converts the generated declarative config to a model and validates it.
+func ValidateFBC(declcfg *declarativeconfig.DeclarativeConfig) error {
```

Comment:
I moved this function from `install.go` (in the run bundle implementation) to `index_image.go`.

These functions are used during both the run bundle installation and the run bundle-upgrade process. Since `install.go` already imports some functionality from `index_image.go`, I moved the common functions to `index_image.go` and exported them to be used in `install.go`.

Reply:
Yeah, the code would look more elegant that way. Do we want to do that now or in a follow-up?